connectors/src/connectors/slack/bot.ts

```diff
@@ -245,37 +262,55 @@ async function botAnswerMessage(
   if (mentions.length === 0) {
     mentions.push({ assistantId: "dust", assistantName: "dust" });
   }
-  // for now we support only one mention.
-  mentions = mentions.slice(0, 1);
-  const convRes = await dustAPI.createConversation({
-    title: null,
-    visibility: "unlisted",
-    message: {
-      content: message,
-      mentions: mentions.map((m) => {
-        return { configurationId: m.assistantId };
-      }),
-      context: {
-        timezone: slackChatBotMessage.slackTimezone || "Europe/Paris",
-        username: slackChatBotMessage.slackUserName,
-        fullName:
-          slackChatBotMessage.slackFullName ||
-          slackChatBotMessage.slackUserName,
-        email: slackChatBotMessage.slackEmail,
-        profilePictureUrl: slackChatBotMessage.slackAvatar || null,
-      },
+
+  const messageReqBody = {
+    content: message,
+    mentions: mentions.map((m) => {
+      return { configurationId: m.assistantId };
+    }),
+    context: {
+      timezone: slackChatBotMessage.slackTimezone || "Europe/Paris",
+      username: slackChatBotMessage.slackUserName,
+      fullName:
+        slackChatBotMessage.slackFullName || slackChatBotMessage.slackUserName,
+      email: slackChatBotMessage.slackEmail,
+      profilePictureUrl: slackChatBotMessage.slackAvatar || null,
     },
-  });
+  };
+
+  let conversationId: string | undefined = undefined;
+  if (lastSlackChatBotMessage?.conversationId) {
+    conversationId = lastSlackChatBotMessage.conversationId;
+    slackChatBotMessage.conversationId = conversationId;
+    await slackChatBotMessage.save();
+    const mesasgeRes = await dustAPI.postUserMessage({
+      conversationId: conversationId,
+      message: messageReqBody,
+    });
+    if (mesasgeRes.isErr()) {
+      return new Err(new Error(mesasgeRes.error.message));
+    }
+  } else {
+    const convRes = await dustAPI.createConversation({
+      title: null,
+      visibility: "unlisted",
+      message: messageReqBody,
+    });
+    if (convRes.isErr()) {
+      return new Err(new Error(convRes.error.message));
+    }
 
-  if (convRes.isErr()) {
-    return new Err(new Error(convRes.error.message));
+    const conversation = convRes.value;
+    conversationId = conversation.sId;
+    slackChatBotMessage.conversationId = conversationId;
+    await slackChatBotMessage.save();
   }
 
-  const conversation = convRes.value;
-
-  slackChatBotMessage.conversationId = conversation.sId;
-  await slackChatBotMessage.save();
-
+  const conversationRes = await dustAPI.getConversation({ conversationId });
```

Comment:
You don't need to getConversation if you createConversation. Only do so if you postUserMessage

Reply:
done